Update to new g4edge refactor and to v1.3.5

stewartboogert opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

stewartboogert committed:
Update to v1.2.4

stewartboogert committed:
bump build number

stewartboogert committed:
ok should not have bumped

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.28.06.15.53

stewartboogert committed:
Remove yum requirements

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.28.06.15.53

stewartboogert committed:
linter suggestion

stewartboogert committed:
Put back yum requirements

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.30.02.36.09

stewartboogert committed:
Include matplotlib

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.30.02.36.09

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.30.02.36.09

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.09.30.02.36.09

stewartboogert committed:
update

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.10.01.11.40.29

stewartboogert committed:
matplotlib-base

stewartboogert committed:
add osx_arm64

stewartboogert committed:
fix arm64 build

stewartboogert committed:
fix arm64 build

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.10.01.11.40.29

stewartboogert committed:
build arm64 on intel

stewartboogert committed:
Remove versions on noarch packages

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.10.01.11.40.29

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.10.01.11.40.29

stewartboogert committed:
MNT: Re-rendered with conda-build 24.7.1, conda-smithy 3.40.1, and conda-forge-pinning 2024.10.01.11.40.29

stewartboogert committed:
Remove python version

stewartboogert committed:
Add osx-arm64 with default provider

stewartboogert committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.41.1, and conda-forge-pinning 2024.10.01.17.37.14

stewartboogert committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.13.01.40

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

stewartboogert committed:
MNT: Re-rendered with conda-build 24.9.0, conda-smithy 3.51.1, and conda-forge-pinning 2025.08.22.13.01.40

stewartboogert committed:
intel cross compilation

stewartboogert committed:
remove native arm64

stewartboogert committed:
Add setuptools for test

gipert commented on conda-forge.yml:
```diff
@@ -1,14 +1,14 @@
-  osx_arm64: osx_64
+    linux_64: linux_64
+    osx_64: osx_64
+    osx_arm64: osx_64
```
@stewartboogert why this? can't we natively compile on arm? (see osx build failure)

gipert replied:
Ok, I see now that you already gave the native build and then cross compilation a go. I don't know how we can get such an "bad cpu type" error if we compile natively on ARM

stewartboogert replied:
In general arm64 apple are cross compiled on intel. So specific runners are required to natively compile on arm64 and I am not sure of the process of getting access. In general the cross compilation works well, I have done it for my large Geant4 based application. Not sure why it is not working for pyg4ometry.

h-vetinari replied:
We are still waiting (yes, even now) for native `osx-arm64` runners in azure pipelines. So currently cross-compilation is the only option for `osx-arm64` builds.

gipert committed:
add maintainers @stewartboogert and @gipert

gipert commented:
@conda-forge-admin please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to rerender for you, but it looks like I wasn't able to push to the `v1.3.5` branch of `g4edge`/`pyg4ometry-feedstock`. Did you check the "Allow edits from maintainers" box?

**NOTE**: Our webservices cannot push to PRs from organization accounts or PRs from forks made from organization forks because of GitHub permissions. Please fork the feedstock directly from conda-forge into your personal GitHub account.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17458084459](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17458084459). Examine the logs at this URL for more detail.</sub>

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert committed:
add build section

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert committed:
add gmp mpfr

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert committed:
try again

gipert commented:
Seems like I fixed it. @conda-forge/core could you please have a look and squash-merge, if happy?

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -1,88 +1,87 @@
 
 extra:
   recipe-maintainers:
-    - chrisburr
```
Why are you removing an existing maintainer?

gipert replied:
I will add it back.

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -1,88 +1,87 @@
-    - sympy >=1.7
-    - scipy
+    - testtools
+    - libgl-devel  # [linux]
```
Is this really necessary at runtime? Normally you only need libgl to build.

gipert replied:
Right, I'll remove

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -1,88 +1,87 @@
+    - antlr4-python3-runtime ==4.13.1
     - matplotlib-base
+    - networkx >=3
+    - {{ pin_compatible('numpy') }}
```
Please remove this line, it's handled by a run-export form numpy

h-vetinari commented on recipe/meta.yaml:
```diff
@@ -1,88 +1,87 @@
-    - openjdk
+    - python                              # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - {{ pin_compatible('numpy') }}       # [build_platform != target_platform]
```
if you need numpy to cross-compile, just do
```suggestion
    - numpy                               # [build_platform != target_platform]
```

h-vetinari commented on conda-forge.yml:
```diff
@@ -1,14 +1,14 @@
-  linux_aarch64: default
-  linux_ppc64le: default
+  linux_64: default
+  osx_64: default
```
the provider settings for linux_64 and osx_64 are unnecessary; you can remove this section entirely if you want (unless you want to build `linux-{aarch64,ppc64le}` in emulation)

h-vetinari commented on conda-forge.yml:
```diff
@@ -1,14 +1,14 @@
 build_platform:
-  linux_aarch64: linux_64
-  linux_ppc64le: linux_64
-  osx_arm64: osx_64
```
You are removing aarch and ppc builds through this change - why?

Also, keep the indentation homogeneous with the rest of the file (2 spaces, as before).

gipert replied:
build failures:

`aarch64`: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1328839&view=logs&j=5b44946f-b0b5-570f-e735-950feabb4722&t=d53332be-ec07-565d-9b53-75513afd6eda&l=4690 -> not VTK wheels for 3.13 apparently

`ppc64le` with python 3.13: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1328839&view=logs&j=f2822e16-0094-5d6c-29fd-a9f5a365b7fd&t=c158e018-8e8e-5c27-0ba4-2482c0a9e77e&l=1350 -> I don't really know how to fix this.

I skipped these builds for now. Happy to take suggestions in case you know how to fix!

gipert committed:
implement suggestions from code review

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert commented:
Thank you for the fast response @h-vetinari! I have implemented your suggestions.

gipert committed:
skip flaky builds

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert committed:
remove aarch64 and ppc64le

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

gipert commented:
@h-vetinari this is good to go from my side.

h-vetinari commented:
The issue with
```
libGLdispatch.so.0: cannot allocate memory in static TLS block
```
is not a reason to skip the builds, cf. [this](https://github.com/conda-forge/conda-forge.github.io/issues/1551). To avoid thinking about it much further, we could do
```yaml
# issues with TLS allocation order, c.f. https://github.com/conda-forge/conda-forge.github.io/issues/1551
c_stdlib_version:  # [aarch64]
  - 2.34           # [aarch64]
```
in `conda-forge.yml` and then rerender. Having builds constrained to recent glibc is still better than removing a target completely. If someone else needs aarch support with older glibc, they can open an issue and it can be investigated independently.

For ppc, the situation is slightly different; we don't have current qt builds for ppc, which means that recent vtk is not available for ppc either, and thus missing for py3.13 entirely. Still, rather than removing it outright for now, I would just add
```yaml
build:
  # no recent qt builds for ppc
  skip: true  # [ppc64le and py==313]
```
in `meta.yaml`, and keep building for python<3.13 for now.

gipert committed:
apply suggestions from code review

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ You are setting `c_stdlib_version` below the current global baseline in conda-forge (10.13). If this is your intention, you also need to override `MACOSX_DEPLOYMENT_TARGET` (with the same value) locally.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17488240575](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17488240575). Examine the logs at this URL for more detail.</sub>

gipert committed:
wrong config file?

gipert committed:
please linter?

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

gipert commented:
@h-vetinari I am not entirely sure the conda config is correct...

h-vetinari commented:
> @h-vetinari I am not entirely sure the conda config is correct...

It's just that the linter is a bit stupid. Let me try a better fix

h-vetinari commented:
Aha, you seem to have unchecked the "allow maintainers to make commits" setting for this PR, so I cannot push.

h-vetinari commented on recipe/conda_build_config.yaml:
```diff
@@ -0,0 +1,6 @@
+c_stdlib_version:  # [osx]
+  - 10.13          # [osx]
+
+# issues with TLS allocation order, c.f. https://github.com/conda-forge/conda-forge.github.io/issues/1551
+c_stdlib_version:  # [aarch64]
+  - 2.34           # [aarch64]
```
```suggestion
# issues with TLS allocation order, c.f. https://github.com/conda-forge/conda-forge.github.io/issues/1551
c_stdlib_version:  # [linux and aarch64]
  - 2.34           # [linux and aarch64]
```

h-vetinari commented on conda-forge.yml:
```diff
@@ -1,14 +1,19 @@
+  # linux_aarch64: linux_64
+  # linux_ppc64le: linux_64
+  osx_64: osx_64
   osx_arm64: osx_64
```
no need to specify native build platforms; also, re-enable aarch/ppc
```suggestion
  linux_aarch64: linux_64
  linux_ppc64le: linux_64
  osx_arm64: osx_64
```

gipert replied:
sorry 🤦🏻

gipert committed:
Update recipe/conda_build_config.yaml

Co-authored-by: h-vetinari <h.vetinari@gmx.com>

gipert committed:
Update conda-forge.yml

Co-authored-by: h-vetinari <h.vetinari@gmx.com>

gipert commented:
>Aha, you seem to have unchecked the "allow maintainers to make commits" setting for this PR, so I cannot push.

I can't see the option so I guess only @stewartboogert can.

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.04.02.19.31

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

h-vetinari commented:
Hm, the error here makes no sense to me.
```
  File "$PREFIX/lib/python3.13/site-packages/pyg4ometry/pycgal/__init__.py", line 1, in <module>
    from . import CGAL
ImportError: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by $PREFIX/lib/python3.13/site-packages/pyg4ometry/pycgal/CGAL.cpython-313-aarch64-linux-gnu.so)
```

We're in an alma9 image with glibc 2.34, so that should definitely contain the 2.32 symbols
```
++ conda info
  [...]
       virtual packages : __archspec=1=zen2
                          __conda=25.5.1=0
                          __glibc=2.34=0         # <----
                          __linux=6.11.0=0
                          __unix=0=0
```

gipert commented:
Any idea on what else we could try?

h-vetinari commented:
Maybe QEMU isn't emulating all the newer glibc symbols yet? Not sure how the compatibility level there gets determined 🤔

We haven't updated QEMU in a while because that's part of our docker images themselves, and there were some major regressions in v9 that made it unusable for us (e.g. numpy/scipy timing out), presumably because QEMU started emulating more advanced instruction sets, but at the cost of greatly increased runtime. Haven't had time to research whether that's the case nor if/how we could switch that off.

For this feedstock, we could try using glibc 2.28 instead of 2.34, which should at least have some partial fixes for the TLS allocation stuff, compared to 2.17.

CC @isuruf

gipert committed:
pin glibc to 2.28

gipert committed:
MNT: Re-rendered with conda-smithy 3.52.1 and conda-forge-pinning 2025.09.07.02.24.17

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

isuruf commented:
Should be fixed by https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/403

h-vetinari reviewed (approved):
Thanks for the patience and persistence here!

> @conda-forge/core could you please have a look and squash-merge, if happy?

LGTM

h-vetinari commented:
Thanks Isuru! CI here passes with 2.28 (perhaps already profiting from the PR you referenced), so this doesn't need 2.34 for now.

gipert commented:
Thank you for the support!
